docs/DesignDocs/Attestation_API_Proposal.md

```diff
@@ -0,0 +1,534 @@
+  helper functions are not part of the OE SDK public API.
+- Remove legacy artifacts that are redundant, e.g. the SGX
+local / remote flags.
+- Streamline design for support of legacy SGX APIs `oe_get_report()` and
+`oe_verify_report()`.
+
+### Proposed Attester Plugin API
+
+Use cases for the OE SDK framework to interact with an attester plugin, supported
+by the attester plugin entry points defined in the `oe_attester_t` structure:
+- Execute a function when the plugin is registered.
+  - The same entry point definition as in the OE SDK V0.9 release.
+- Execute a function when the plugin is unregistered:
+  - The same entry point definition as in the OE SDK V0.9 release.
+- Get an evidence in an globally unique format, optionally along with a set of
+endorsements.
+  - Entry point `oe_result_t (*get_evidence)(oe_attester_t* context, const oe_claim_t* custom_claims, size_t custom_claims_length, const void* opt_params, size_t opt_params_size, uint8_t** evidence_buffer, size_t* evidence_buffer_size, uint8_t** endorsements_buffer, size_t* endorsements_buffer_size)`.
+  - The legacy `flags` parameter in the OE SDK V0.9 release is removed.
+- Get a legacy-format report
+  - Entry point `oe_result_t (*get_report)(...)`
+  - Note: this entry point must be implemented if the OS SDK framework
+  needs to support its legacy public API `oe_get_report()` using this plugin.
+- Free a dynamically allocated evidence buffer.
+  - The same entry point definition as in the OE SDK V0.9 release.
+- Free a dynamically allocated endorsements buffer.
+  - The same entry point definition as in the OE SDK V0.9 release.
+
+The OE SDK framework exposes a set of helper functions for attester plugin
+libraries to interact with it. These helper functions are for plugin library
+developers, and not part of the public API for application developers.
+
+Below are the use cases for an attester plugin library to interact with the
+OE SDK framework:
+- Register an attester plugin.
+  - Helper function
+  `oe_result_t oe_register_attester(oe_attester_t* plugin, const void* config_data, size_t config_data_size)`.
+  - The same definition as in the OE SDK V0.9 release, except that this function
+  is not part of the public API.
+  - The optional `config_data`, e.g. for holding platform-specific configuration,
+  is provided by the plugin library, not by the application as in the V0.9
+  release.
+- Unregister an attester plugin.
+  - Helper function `oe_result_t oe_unregister_attester(oe_attester_t* plugin)`.
+  - The same definition as in the OE SDK V0.9 release, except that this function
+  is not part of the public API.
+
+In the OE SDK V0.10 release, attester plugins are managed internally between
+the OE SDK framework and plugin libraries, without exposing details to
+application software. As explained in the
+[[Design Notes]](https://github.com/shnwc/openenclave/blob/master/docs/DesignDocs/NotesOnAttestationAPI.md#options-for-enclave-side-plugin-library-initialization)
```

Comment:
In the NotesOnAttestationAPI.md, option 1, 2, and 4 have potential issue.  Option 1 and 2 do not support multiple external plugins. Option 4 is not guaranteed to work universally on all platforms and configurations. Option 3 is a potential solution which issue 2093 is trying to resolve.

Without a proper plugin initialization solution, I'd suggest we either don't change current initialization design or make it clear to the users that such design is purely 'experimental'.

Reply:
Yes, let's mark the plugin design as "experimental", until issue #2903 is resolved.

Reply:
I'll add text to indicate that the proposed API and plugin design continue be marked as  "experimental"